### PR TITLE
Support batch inference of Mask R-CNN

### PR DESCRIPTION
**Pull request by hhaAndroid**

### CODE REVIEW

File: mmdet/models/roi_heads/test_mixins.py [hunk 1]
```diff
@@ -253,58 +253,67 @@ def simple_test_mask(self,
...
+                                       *mask_pred.shape[1:])
+
+        # apply mask post-processing to each image individually
+        segm_results = []
+        for i in range(batch_size):
+            mask_pred = mask_preds[i]
+            det_bbox = det_bboxes[i]
+            det_label = det_labels[i]
+
+            supplement_mask = det_bbox[..., -1] != 0
```
Comment (Johnson-Wang) [inline mmdet/models/roi_heads/test_mixins.py:303]: Add a comment here.

**Comment by RunningLeon** [comment]
@hhaAndroid Works good for mask-rcnn exporting to ONNX

**Review by RunningLeon — APPROVED** [review]
LGTM

**Review by Johnson-Wang — APPROVED** [review]

File: mmdet/models/roi_heads/test_mixins.py [hunk 2]
```diff
@@ -253,58 +253,68 @@ def simple_test_mask(self,
...
+        batch_size = det_bboxes.size(0)
+        num_proposals_per_img = det_bboxes.shape[1]
+
+        # if det_bboxes is rescaled to the original image size, we need to
+        # rescale it back to the testing scale to obtain RoIs.
+        if rescale and not isinstance(scale_factors[0], float):
+            scale_factors = det_bboxes.new_tensor(scale_factors)
+
+        det_bboxes = det_bboxes[..., :4]
+        if rescale:
```
Comment (ZwwWayne) [inline mmdet/models/roi_heads/test_mixins.py:281]: Put the logic of rescale together for a more clear logic. For example
```python
det_bboxes = det_bboxes[..., :4]
if rescale:
    if not isinstance(scale_factors[0], float):
        scale_factors = det_bboxes.new_tensor(scale_factors)
    det_bboxes = det_bboxes * scale_factor.unsqueeze(-1)

```

**Pull request merged**